WordPress/Classes/ViewRelated/NUX/UIView+Margins.swift

```diff
@@ -0,0 +1,24 @@
+import UIKit
+
+extension UIView {
+
+    func getHorizontalMargin(compactMargin: CGFloat = 0.0) -> CGFloat {
+        guard traitCollection.verticalSizeClass == .regular,
+              traitCollection.horizontalSizeClass == .regular else {
+                  return compactMargin
+              }
+
+        let isLandscape = UIDevice.current.orientation.isLandscape
+        let multiplier: CGFloat = isLandscape ? .ipadLandscape : .ipadPortrait
+
+        return frame.width * multiplier
+    }
+
+}
+
+private extension CGFloat {
+
+    static let ipadPortrait: CGFloat = 0.1667
+    static let ipadLandscape: CGFloat = 0.25
```

Comment:
I'm quite not sure if I understand these numbers.

Are these in order to properly set up the margins in iPad? If so you can probably rely on [readableContentGuide](https://developer.apple.com/documentation/uikit/uiview/1622644-readablecontentguide).

Reply:
These values were being pulled from the code I was refactoring:

- https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/NUX/LoginEpilogueViewController.swift#L207-L210
- https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/NUX/SignupEpilogueViewController.swift#L320-L323

Using the readable content guide is a good idea. That would let us use the layout builder instead of manually setting these values. I made a commit with those changes here: https://github.com/wargcm/WordPress-iOS/commit/72f25726ae4e4400f3f69b97755463ed3f114e89

I didn't include it in this PR because I wanted to check with you on something. There are separate view controllers in a pod that calculates the margins the same way that `LoginEpilogueViewController` and `SignupEpilogueViewController` does. They're located here:

- https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticator/Signin/LoginPrologueViewController.swift#L592-L618
- https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticator/Signin/LoginViewController.swift#L374-L400

What are your thoughts about pulling that commit in? I could update the two view controllers above to also use the readable margins so they would all align properly.

Reply:
Oh, I didn't realize this came from the current login flow/lib.

I think we can make this change in another PR then. But in order to help us decide if we should do it, let's double-check why it is the way it is. :)

@ScoutHarris there's any specific reason why we relied on the constants mentioned in the comment above instead of using [readableContentGuide](https://developer.apple.com/documentation/uikit/uiview/1622644-readablecontentguide)?

Reply:
Uh.... it's been a while, but I'm guessing because I just didn't think to use the `readableContentGuide`. As long as the button widths remain the same in both orientations, do what you like. 😄